### PR TITLE
Tighten text contrast bounds, fall back to larger bounds

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes the issues outlined in https://github.com/flutter/flutter/pull/100267

Historically, we implemented text contrast checking assuming that we were only really looking at text color in comparison to a larger background surface. Thus it made sense to expand the area sampled from slightly to ensure that we picked up enough of this color - especially considering that flutter test will use ahem by default which is quite large and blocky.

Nevertheless, there are some cases where this doesn't work well -  such as through the usage of the text background color which is _extremely_ tight to the text bounds themselves.

To adjust for this, we can expand the heuristic to first consider tighter bounds, and then if it fails to find multiple colors or only finds colors that extremely close, fall back to the historical larger bounds.

This also takes an opportunity to correct two instances of `>` that should be `>=` when determining minimum text size

Example of AA on Ahem, see the very last set of boxes:
foo_bar_bazz

### CODE REVIEW

File: packages/flutter_test/test/accessibility_test.dart [hunk 1]
```diff
@@ -126,9 +126,9 @@ void main() {
       expect(result.reason,
         'SemanticsNode#4(Rect.fromLTRB(300.0, 200.0, 500.0, 400.0), label: "this is a test",'
         ' textDirection: ltr):\nExpected contrast ratio of at least '
-        '4.5 but found 1.17 for a font size of 14.0. The '
-        'computed light color was: Color(0xfffafafa), The computed dark color was:'
-        ' Color(0xffffeb3b)\n'
+        '4.5 but found 0.88 for a font size of 14.0. The '
```
Comment (jonahwilliams) [inline packages/flutter_test/test/accessibility_test.dart:129]: These are the correct colors

**Comment by jonahwilliams** [comment]
This may have difficulty rolling into google3, if we were incorrectly passing some instances before. Nevertheless, I'll have to investigate these failures since it could just be failures or the new heuristic.

**Comment by jonahwilliams** [comment]
actually this won't quite work with text-aa. Maybe instead I should check if there is a background color on the text object (since there is no other way for that to get painted) and select the rect based on that

**Review by dnfield — APPROVED** [review]

**Comment by jonahwilliams** [comment]
(actually investigating more, this may be working correctly)....

File: packages/flutter_test/lib/src/accessibility.dart [hunk 2]
```diff
@@ -482,34 +496,49 @@ class _ContrastReport {
         lightCount = count;
       }
     }
+    // Sometimes the selected colors are imperceptibly close, due to picking up
+    // one primary color and a few anti-aliased pixels. If this is the case, then
+    // mark areColorsClose to true so the tester can expand the selection area.
+    final Color lightColorObject = Color(lightColor);
+    final Color darkColorObject = Color(darkColor);
+    const double kMinimumDistance = 10;
```
Comment (jonahwilliams) [inline packages/flutter_test/lib/src/accessibility.dart:504]: This is somewhat arbitrary based on the raised button example

File: packages/flutter_test/lib/src/accessibility.dart [hunk 3]
```diff
@@ -234,28 +234,30 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
       // contrast ratio based on text size/weight.
       double? fontSize;
       bool isBold;
+      // Italic ahem font has AA artifacts between characters and requires an expanded rect.
+      bool forceLargeRect = false;
```
Comment (jonahwilliams) [inline packages/flutter_test/lib/src/accessibility.dart:238]: I did find one issue, is that we hit AA artifacts on ahem if it is itallic. We should investigate a way to disable AA globally for these sorts of tests

**Pull request closed without merging**